entity/node/accessors/node_modules/iotAuthService.js

```diff
@@ -325,6 +325,13 @@ function sendSessionKeyReqHelper(helperOptions, helperEventHandlers, options, se
         console.log('received auth hello!');
         var obj = parseAuthHello(helperOptions.payload);
         console.log(obj);
+        console.log(obj.authId);
+        console.log(options);
+        if (obj.authId === options.authId) {
+            console.log('auth id matches');
+        } else {
+            throw 'auth id mismatch';
```

Comment:
Can you follow regular error handling for this function and change this to `return {success: false, error: '...'}`?